packages/grid/x-data-grid-pro/src/tests/columns.DataGridPro.test.tsx

```diff
@@ -420,4 +421,23 @@ describe('<DataGridPro /> - Columns', () => {
       ]);
     });
   });
+
+  it.only('should throw if unsupported column type is used', function test() {
```

Comment:
Could you add where we add a custom `str` column type and check it does not crash ?